sympy/sets/sets.py

```diff
@@ -218,16 +218,12 @@ def _complement(self, other):
 
         elif isinstance(other, FiniteSet):
             from sympy.utilities.iterables import sift
+            from sympy.core.logic import fuzzy_bool
 
-            def ternary_sift(el):
-                contains = self.contains(el)
-                return contains if contains in [True, False] else None
-
-            sifted = sift(other, ternary_sift)
+            sifted = sift(other, lambda x: fuzzy_bool(x))
```

Comment:
You can directly pass `fuzzy_bool` function like `sifted = sift(other, fuzzy_bool)`